### PR TITLE
tools: add mavfft_pid.py to allow FFT analysis for pids in log files

### PR DESCRIPTION
**Pull request by andyp1per**
Example output

image

FYI @lthall

### CODE REVIEW

File: tools/mavfft_pid.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        for ax in ['P', 'I', 'D']:
+            data[gyr+'.'+ax] = []
+
+    # now gather all the data
+    while True:
+        m = mlog.recv_match(condition=args.condition)
+        if m is None:
+            break
+        type = m.get_type()
+        if type.startswith("PIDR"):
```
Comment (peterbarker) [inline tools/mavfft_pid.py:42]: Why not direct equivalence?  Could be rather faster if there's a checksum associated with the strings, for example.

File: tools/mavfft_pid.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            data[type+'.P'].append(m.P)
+            data[type+'.I'].append(m.I)
+            data[type+'.D'].append(m.D)
+        if type.startswith("PIDP"):
```
Comment (peterbarker) [inline tools/mavfft_pid.py:46]: `elif` here will make things faster.

File: tools/mavfft_pid.py (hunk 1 above)
Comment (andyp1per) [reply tools/mavfft_pid.py]: My python is like my french - limited and broken. I tend to stick with things I've copied and pasted. Anyway fixed now.

File: tools/mavfft_pid.py (hunk 2 above)
Comment (andyp1per) [reply tools/mavfft_pid.py]: Fixed

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**